src/borg/archiver.py

```diff
@@ -2422,6 +2422,17 @@ def main():  # pragma: no cover
         tb_log_level = logging.ERROR if important else logging.DEBUG
         tb = sysinfo()
         exit_code = EXIT_ERROR
+    except OSError as e:
+        at_frame = traceback.extract_tb(e.__traceback__, limit=-1)[0]
+        msg = textwrap.dedent("""
+        I/O error: {error}
+
+        At: {frame.filename}:{frame.lineno} ({frame.name})
```

Comment:
yes, this "RFC" is about how we want to balance the error reporting here. That's a question of gauging probabilities plus some "UX".

- If there are IO errors (actual EIO, or permission issues, or missing files, full disks, ...) the user wants to know where (in the FS / which disk)
  - But only a select few exceptions actually have enough information for telling that (practically only errors on open())
  - The next best thing would be (if somewhat convoluted) telling where (in the code) ("At:" ...)
  - What's kinda bad is throwing a long traceback that's not readily understood and tends to obscure the error (#1287 is a good example, the actual error is somewhere in the middle, followed and preceded by dozens of lines).

- If there are bugs *we* want to know if/how/where they happen in the code. I would argue that in most cases the traceback isn't all that important, because most likely it is incorrect application of API to an object (#1287 is a fine specimen of this), which I would say is usually clear from the location in the code itself. State errors were not OSErrors in the past (writing|flushing|reading... a closed file is a ValueError for example).
- There is also the question assuming that we have a bug that caused this specific error and is *not* reproducible at all (and actually a bug in borg and not a problem in a lower layer), how likely it is that the traceback contains useful information that helps analyse it beyond this. In the past, with other errors, a traceback wasn't sufficient at times, since it didn't have important information (like locals).

  So I'm not sure whether that's a probable case we need to consider.